File: UI/src/app/core/models/customer.ts
export interface Customer {
  id: number;        // .NET int -> TS number
  name: string;      // .NET string -> TS string
  email: string;     // .NET string -> TS string
  balance: number;   // .NET decimal/double -> TS number
  createdAt: string; // .NET DateTime -> TS string (ISO Format)
}

File: IlkProjem.Core/Models/Customer.cs
namespace IlkProjem.Core.Models;

public class Customer : BaseEntity
{
    public int Id { get; set; } 
    public required string Name { get; set; }
    public string? Email { get; set; }
    public decimal Balance { get; set; }

    // --- DOSYA BAĞLANTISI BURADA BAŞLIYOR ---
    
    // 1. Foreign Key: Dosyanın Guid'ini burada tutacağız
    public Guid? ProfileImageId { get; set; } 

    // 2. Navigation Property: Kod tarafında direkt "customer.ProfileImage" diyebilmek için
    public virtual Files? ProfileImage { get; set; }

    public virtual ICollection<House> Houses { get; set; } = new List<House>();
    public virtual ICollection<Car> Cars { get; set; } = new List<Car>();
}